engines/buried/sound.cpp

```diff
@@ -0,0 +1,761 @@
+			_soundData[kAmbientIndexBase + _lastAmbient]->_timedEffectStart = g_system->getMillis();
+			_soundData[kAmbientIndexBase + _lastAmbient]->_timedEffectRemaining = 2000;
+	
+			// Reset parameters for the current ambient
+			g_system->getMixer()->setChannelVolume(*_soundData[kAmbientIndexBase + _lastAmbient]->_handle,
+					clipVolume(_soundData[kAmbientIndexBase + _lastAmbient]->_volume << 1));
+		}
+
+		// Load the new ambient
+		if (_soundData[kAmbientIndexBase + newAmbientTrack]->load(fileName)) {
+			// Set the parameters of the new ambient
+			_soundData[kAmbientIndexBase + newAmbientTrack]->_volume = 0;
+			_soundData[kAmbientIndexBase + newAmbientTrack]->_loop = true;
+			_soundData[kAmbientIndexBase + newAmbientTrack]->_timedEffectIndex = TIMED_EFFECT_VOLUME;
+			_soundData[kAmbientIndexBase + newAmbientTrack]->_flags = 0;
+			_soundData[kAmbientIndexBase + newAmbientTrack]->_timedEffectSteps = 16;
+			_soundData[kAmbientIndexBase + newAmbientTrack]->_timedEffectDelta = finalVolumeLevel / 16;
+			_soundData[kAmbientIndexBase + newAmbientTrack]->_timedEffectStart = g_system->getMillis();
+			_soundData[kAmbientIndexBase + newAmbientTrack]->_timedEffectRemaining = 2000;
+
+			// Start the new ambient
+			retVal = _soundData[kAmbientIndexBase + newAmbientTrack]->start();
+		}
+	} else {
+		// Load the new ambient
+		if (!fileName.empty()) {
+			if (_soundData[kAmbientIndexBase + newAmbientTrack]->load(fileName)) {
+				// Set some parameters
+				_soundData[kAmbientIndexBase + newAmbientTrack]->_volume = finalVolumeLevel;
+				_soundData[kAmbientIndexBase + newAmbientTrack]->_loop = true;
+
+				// Stop the current ambient
+				delete _soundData[kAmbientIndexBase + _lastAmbient];
+				_soundData[kAmbientIndexBase + _lastAmbient] = new Sound();
+
+				// Start the new ambient
+				retVal = _soundData[kAmbientIndexBase + newAmbientTrack]->start();
+			}
+		} else {
+			// Stop the current ambient
+			delete _soundData[kAmbientIndexBase + _lastAmbient];
+			_soundData[kAmbientIndexBase + _lastAmbient] = new Sound();
+		}
+	}
+
+	// Reset the last ambient index
+	_lastAmbient = newAmbientTrack;
+
+	// Return success
+	return retVal;
+}
+
+bool SoundManager::adjustAmbientSoundVolume(byte newVolumeLevel, bool fade, byte steps, uint32 fadeLength) {
+	// If we are not playing an ambient track, simply return false
+	if (!_soundData[kAmbientIndexBase + _lastAmbient]->_handle)
+		return false;
+
+	// Compare the new volume level to the current one, returning success if they are the same
+	if (_soundData[kAmbientIndexBase + _lastAmbient]->_volume == newVolumeLevel)
+		return true;
+
+	// If we already have any timed channels in the current ambient channel, kill them now
+	if (_soundData[kAmbientIndexBase + _lastAmbient]->_timedEffectIndex != TIMED_EFFECT_NONE) {
+		_soundData[kAmbientIndexBase + _lastAmbient]->_timedEffectIndex = TIMED_EFFECT_NONE;
+		_soundData[kAmbientIndexBase + _lastAmbient]->_flags = 0; // clone2727 says: is this right?
```

Comment:
This is a good candidate for replacing with a "TODO: Is this right?". That way, it will be easier to remember that it's something still to be looked into.